src/content/en/2020/fonts.md

```diff
@@ -22,3 +22,266 @@ queries: 04_Fonts
+  )
+}}
+
+{{ figure_markup(
+  image="fonts-resource-hints-performance-mobile.png",
+  caption="resource-hints performance, mobile.",
+  description="A bar chart showing the mobile median first content paint and last content paint (in milliseconds) for different resource-hints settings.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vT2Q4hcDGGdclJH2ym0Pp_f8JWvYur_OQFQNkuScJyO7_ZCR1KPZsewL-mEZhxcuRFcde_Mxio8z_8P/pubchart?oid=448614843&format=interactive",
+  sheets_gid="655527383",
+  sql_file="04_07.font_resource_hints_with_fcp.sql"
+  )
+}}
+
+Analysis of this data suggests that the `dns-prefetch` setting, while the most popular, doesn't improve performance much. Presumably, the DNS for popular web font servers are likely to be cached anyway. The other settings give a lot more bang for the buck, with `preconnect` being a sweet spot for ease of use, flexibility, and performance improvement. As of March 2020, Google Fonts recommends adding this line to the HTML source, immediately before the CSS link:
+
+```html
+<link rel="preconnect" href="https://fonts.gstatic.com">
+```
+
+The use of `preconnect` has grown considerably since last year, now at 8% from 2%, but there's a lot more potential performance still left on the table. Adding this line might be the single best optimization for web pages that use Google Fonts.
+
+It might be tempting to go even farther into the pipeline, preloading or prerendering the font asset, but that potentially conflicts with other optimizations, such as fine-tuning the font for the capabilities of the rendering engine, or the `unicode-range` optimization described below. To preload a resource, you have to know *exactly* what resource to load, and the best resource for the task may depend on information not readily available at HTML authoring time.
+
+## Home on the (Unicode) range
+
+Fonts increasingly have support for lots and lots of languages. Other fonts can have a large number of glyphs because the script (especially CJK) requires it. Either way can increase the file size. That's unfortunate if the web page is not in fact a multilingual dictionary, and only uses a fraction of the font's capabilities.
+
+One older approach is for the HTML author to explicitly indicate a font subset. However, that requires deeper knowledge of the content, and risks a "ransom note" effect when the content uses characters supported by the font but not by the chosen subset. See the excellent essay [When fonts fall](https://www.figma.com/blog/when-fonts-fall/) by Marcin Wichary for lots more detail about how fallback works.
+
+The `unicode-range` feature is a newer approach to this problem. The font is sliced into subsets, each with a separate `@font-face` rule that indicates the Unicode coverage for that slice with a `unicode-range` descriptor. The browser then analyzes the content as part of its rendering pipeline, and downloads *only* the slices needed to render that content.
```

Comment:
Unicode-range is not at all a new feature. Suggest instead "Static subsets, indicated by `unicode-range`, are a better approach to this problem".

Reply:
Need to also point out the problem with static subsets. Suggest adding, after "that content.", the following:

"For alphabetic languages, this typically works well although it can result in poor kerning between characters in different subsets. For languages which rely on glyph shaping, such as Arabic, Urdu and many Indic languages, static subsets frequently result in broken text rendering. And for CJK, static subsets based on contiguous Unicode ranges provide almost no benefit because the characters used on a particular page are scattered almost randomly across the various subsets."

Reply:
These caveats are true, but it seems to me it could use a conclusion. How about: "Because of these reasons, correct and performant use of static subsets is tricky, and requires careful analysis and implementation."